jmclient/jmclient/wallet_utils.py

```diff
@@ -502,11 +504,11 @@ def get_addr_status(addr_path, utxos, utxos_enabled, is_new, is_internal):
             addr_balance += utxodata['value']
 
             #TODO it is a failure of abstraction here that
-            # the bitcoin core interface is used directly
-            #the function should either be removed or added to bci
-            #or possibly add some kind of `gettransaction` function
+            # the no-history interface is excluded in this way.
+            # the function should either be removed or added to bci
+            # or possibly add some kind of `gettransaction` function
             # to bci
-            if jm_single().bc_interface.__class__ == BitcoinCoreInterface:
+            if not isinstance(jm_single().bc_interface, BitcoinCoreNoHistoryInterface):
```

Comment:
The alternative is `if isinstance(jm_single().bc_interface, BitcoinCoreInterface)` right? Which because of the inheritance would work properly for regtest, rather than what we had before, right?

I think we briefly discussed this. I agree with your option because I think not having tx querying ability (even for one's own wallet) should be specific to a "NoHistoryInterface". Arguable I guess, but this looks good.